core/experiments/wp-cli.php

```diff
@@ -124,13 +124,9 @@ private function is_network( $assoc_args ) {
 	 * @return void
 	 */
 	private function foreach_sites( callable $callback ) {
-		/** @var \WP_Site[] $sites */
-		$sites = get_sites();
-
-		foreach ( $sites as $keys => $site ) {
-			// Cast $blog as an array instead of object
-			$blog_id = $site->blog_id;
+		$blog_ids = get_sites( [ 'fields' => 'ids', 'number' => 0 ] );
```

Comment:
Hi, Thanks for the fix!

Can you please extract it to a method for re-use?
```suggestion
		$blog_ids = $this->get_sites();
```

```
private function get_sites() { 
return get_sites(  [ 'fields' => 'ids', 'number' => 0 ]  );
```
```suggestion
		$blog_ids = get_sites( [ 'fields' => 'ids', 'number' => 0 ] );
```

Reply:
Hi, sorry, must have missed this. I'll look into it.

Reply:
Just looked into it, extracting it for re-use sure seems like a decent thing to do, but I don't think it's in scope of the original issue, I'm simply offering a bugfix here. Refactoring raises more questions and only complicates things (usage is spread over 5 files, and there doesn't seem to be an obvious central place to put this method). 

Can we please get this merged? We're well over 2 months since the initial report and PR.

Reply:
I agree, we can take it from here. 
Thanks for sending it!
I will merge it.